pilot/pkg/xds/eds.go

```diff
@@ -176,7 +171,6 @@ func (s *DiscoveryServer) edsCacheUpdate(clusterID, hostname string, namespace s
 		Name:      hostname,
 		Namespace: namespace,
 	}: {}})
-	ep.mutex.Unlock()
```

Comment:
The cache clear must be in the lock. See the comment above

Reply:
Reverted. But could not understand from the comments. 
```
 a race condition is introduced where an XDS response may be generated before the update, 
but not completed until after a response after the update
```